test/unit/suite-collection.js

```
@@ -48,204 +51,399 @@ describe('suite-collection', function() {
-
-            collection
-                .disableAll()
-                .enable(suite, {state: 'some-state'});
-
-            assertEnabled(suite, someState);
-            assertDisabled(suite, otherState);
-        });
-
-        it('should apply all enable rules', function() {
-            var suite = util.makeSuiteStub(),
-                collection = new SuiteCollection([suite]),
-                someState = util.makeStateStub(suite, {name: 'some-state'}),
-                otherState = util.makeStateStub(suite, {name: 'other-state'});
-
-            collection
-                .disableAll()
-                .enable(suite, {state: 'some-state'})
-                .enable(suite, {state: 'other-state'});
-
-            assertEnabled(suite, someState);
-            assertEnabled(suite, otherState);
-        });
-
-        it('should enable only specified state in specified browser', function() {
-            var suite = util.makeSuiteStub(),
-                collection = new SuiteCollection([suite]),
-                state = util.makeStateStub(suite, {name: 'some-state'});
-
-            collection
-                .disableAll()
-                .enable(suite, {state: 'some-state', browser: 'ie9'});
-
-            assertEnabled(suite, state, 'ie9');
-            assertDisabled(suite, state, 'firefox');
+        describe('enable', function() {
+            it('should enable suite by full name', function() {
+                var tree = mkTree({
+                        parent: {
+                            child: []
+                        }
+                    }, {browsers: ['b1', 'b2']});
+
+                new SuiteCollection([tree.parent])
+                    .disableAll()
+                    .enable(tree.child.fullName);
+
+                assert.deepEqual(tree.child.browsers, ['b1', 'b2']);
+            });
+
+            it('should fail on attempt to disable unknown suite', function() {
```

`should fail on attempt to ENABLE unknown suite`